aspnetcore/migration/1x-to-2x/index.md

```diff
@@ -99,6 +99,65 @@ The adoption of this new 2.0 pattern is highly recommended and is required for p
 Unable to create an object of type '<Context>'. Add an implementation of 'IDesignTimeDbContextFactory<Context>' to the project, or see https://go.microsoft.com/fwlink/?linkid=851728 for additional patterns supported at design time.
 ```
 
+<a name="add-modify-configuration"></a>
+
+## Add or modify configuration providers
+
+In 1.x projects, adding configuration providers to an application was done during the middleware process by creating an instance of `ConfigurationBuilder`, loading applicable providers (environment variables, app settings, etc), and initializing a a member of `IConfigurationRoot`.
+
+```csharp
+public Startup(IHostingEnvironment env)
+    {
+        var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+        Configuration = builder.Build();
+    }
+
+public IConfigurationRoot Configuration { get; }        
+
+```
+The above example loads the `Configuration` member with environment variables supplied in *launchSettings.json* and configuration settings from *appsettings.json* as well as any appsettings file that matches the `EnvironmentName` property provided in `IHostingEnvironment`. The location of these files would be at the same path as `Startup.cs`
+
+In 2.0 projects, the boilerplate configuration code inherent to 1.x projects runs behind-the-scenes. For example, environment variables and app settings are loaded at startup. All that is required in *Startup.cs* is `IConfiguration` initialization with the injected instance:
+
+```csharp
+public class Startup
+{
+    public Startup(IConfiguration configuration)
+    {
+        Configuration = configuration;
+    }
+
+    public IConfiguration Configuration { get; }
+```
+
+To add or modify configuration providers, invoke the `IWebHostBuilder.ConfigureAppConfiguration` method in *Program.cs*:
```

Comment:
- You can't modify, you can only add more providers.
- We should also show an example of clearing all previous providers and adding new ones.

Reply:
@davidfowl is there a more elegant way to resetting providers than just recreating `IConfigurationRoot` in *Startup.cs* constructor? Not seeing any extension on `IWebHostBuilder` to clear or remove configurations.